python/cudf_polars/cudf_polars/dsl/expressions/string.py

```diff
@@ -419,6 +425,54 @@ def do_evaluate(
                 ),
                 dtype=self.dtype,
             )
+        elif self.name in {
+            StringFunction.Name.SplitExact,
+            StringFunction.Name.SplitN,
+        }:
+            is_split_n = self.name is StringFunction.Name.SplitN
+            n = self.options[0]
+            child, expr = self.children
+            column = child.evaluate(df, context=context)
+            if n == 1 and self.name is StringFunction.Name.SplitN:
+                plc_column = plc.Column(
+                    self.dtype.plc,
+                    column.obj.size(),
+                    None,
+                    None,
+                    0,
+                    column.obj.offset(),
+                    [column.obj],
+                )
+            else:
+                assert isinstance(expr, Literal)
+                by = plc.Scalar.from_py(expr.value, expr.dtype.plc)
+                # See https://github.com/pola-rs/polars/issues/11640
+                # for SplitN vs SplitExact edge case behaviors
+                max_splits = n if is_split_n else 0
+                plc_table = plc.strings.split.split.split(
+                    column.obj,
+                    by,
+                    max_splits - 1,
+                )
+                children = plc_table.columns()
+                ref_column = children[0]
+                if (remainder := n - len(children)) > 0:
+                    children.extend(
+                        plc.Column.all_null_like(ref_column, ref_column.size())
+                        for _ in range(remainder + int(not is_split_n))
+                    )
```

Comment:
Can you add a comment explaining this bit? Something like "fewer splits than expected so we fill the remaining fields with null"